kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -284,9 +285,13 @@ impl<'tcx> GotocCtx<'tcx> {
                             TagEncoding::Direct => {
                                 // then the scalar field stores the discriminant
                                 let discr_ty = self.codegen_enum_discr_typ(ty);
-
                                 let init = self.codegen_scalar(s, discr_ty, span);
-                                self.codegen_direct_literal(ty, init)
+                                let cgt = self.codegen_ty(ty);
+                                Expr::struct_expr_with_nondet_fields(
```

Comment:
How does this interact with your zero_initializer PR? Is this function covered there?

Reply:
Also, this code only runs for enums without additional data, just the discriminant, correct? Because only the discriminant is initialized. If that's the case, then using `srtuct_expr_with_nondet_fields` is misleading because there are no other fields to initialize with nondet.

Reply:
Added a check for this